volume/driver/overlay_linux.go

```diff
@@ -196,28 +144,149 @@ func (driver *OverlayDriver) workDir(datapath string) string {
 	return filepath.Join(driver.OverlaysDir, "work", driver.getGUID(datapath))
 }
 
-func (driver *OverlayDriver) ancestry(datapath string) ([]string, error) {
-	ancestry := []string{}
+func (driver *OverlayDriver) getGUID(datapath string) string {
+	return filepath.Base(filepath.Dir(datapath))
+}
+
+// NewLiveVolume creates the representation of a volume under the `live`
+// directory.
+//
+// It captures the filepath to the volume, as well as its parents (in a singly
+// linked list).
+//
+// ps.: it DOES NOT protect itself against circular dependencies.
+//
+func NewLiveVolume(root, vol string) (*LiveVolume, error) {
+	volDir := filepath.Join(root, vol)
+	parentSymlink := filepath.Join(volDir, "parent")
+	liveVol := &LiveVolume{Path: volDir}
 
-	currentPath := datapath
-	for {
-		ancestry = append(ancestry, driver.layerDir(currentPath))
+	_, err := os.Stat(volDir)
+	if err != nil {
+		return nil, err
+	}
 
-		parentVolume, err := os.Readlink(filepath.Join(filepath.Dir(currentPath), "parent"))
+	parentDir, err := readlink(parentSymlink)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return liveVol, nil
+		}
+
+		return nil, err
+	}
+
+	liveVol.Parent, err = NewLiveVolume(root, filepath.Base(parentDir))
+	if err != nil {
+		return nil, err
+	}
+
+	return liveVol, nil
+}
+
+// Ancestry traverses LiveVolume linked list producing a list of LiveVolumes
+// from "no dependencies" to "with dependencies".
```

Comment:
nit: from "no dependencies" aka oldest to "with dependencies" aka youngest

Reply:
wdyt?

```diff
- // from "no dependencies" to "with dependencies". 
+ // from "no dependencies" (oldest) to "with dependencies" (newest). 
```